src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs

```diff
@@ -2691,25 +2691,21 @@ public override TAbstractAnalysisValue Visit(IOperation operation, object? argum
                     HandlePossibleThrowingOperation(operation);
                 }
 
-                if (_pendingArgumentsToPostProcess.Any(arg => arg.Parent == operation))
+                var pendingArguments = _pendingArgumentsToPostProcess.ExtractAll(static (arg, operation) => arg.Parent == operation, operation);
+                foreach (IArgumentOperation argumentOperation in pendingArguments)
                 {
-                    var pendingArguments = _pendingArgumentsToPostProcess.Where(arg => arg.Parent == operation).ToImmutableArray();
-                    foreach (IArgumentOperation argumentOperation in pendingArguments)
+                    bool isEscaped;
+                    if (_pendingArgumentsToReset.Remove(argumentOperation))
                     {
-                        bool isEscaped;
-                        if (_pendingArgumentsToReset.Remove(argumentOperation))
-                        {
-                            PostProcessEscapedArgument(argumentOperation);
-                            isEscaped = true;
-                        }
-                        else
-                        {
-                            isEscaped = false;
-                        }
-
-                        PostProcessArgument(argumentOperation, isEscaped);
-                        _pendingArgumentsToPostProcess.Remove(argumentOperation);
```

Comment:
@mavasani In order to use the `ExtractAll` method (which has clear semantics), I needed to hoist this call to `Remove` outside the `foreach` (where it combined with the `Where` method). Considering there is no other code that checks `_pendingArgumentsToPostProcess.Contains`, it seemed like a safe change.